drivers/ntb/hw/amd/ntb_hw_amd.c

```diff
@@ -1239,6 +1244,17 @@ static int amd_ntb_pci_probe(struct pci_dev *pdev,
 	if (rc)
 		goto err_init_dev;
 
+	/*
+	 * If our READY bit in SIDEINFO register is already set, it likely
+	 * means we have crashed and the other controller could miss it.
+	 * In such case clear the bit, interrupt the other side, and give
+	 * it some time to see zero before we set the READY again below.
+	 */
+	if (amd_deinit_side_info(ndev)) {
```

Comment:
just wondering, what would happen if both sides were trying to do this at the same time? (i'm not sure that can occur tho)

Reply:
On the first controller it generally should not happen, since there NTB device is local there and should be reset on reboot.  Be even if it happen somehow, they should just try to exchange the extra interrupts, which should be ignored due to still being masked an that point, both wait for a second and continue as normal.